MetaMask/metamask-mobile
# fix: Send ppom metrics when transaction is cancelled.

## **Description**

We should send PPOM/Blockaid metrics parameters when a transaction from wallet is rejected. Right now we only send the metrics when the transaction is approved.

## **Related issues**

Fixes: [#8824](https://github.com/MetaMask/metamask-mobile/issues/8824)

## **Manual testing steps**

1. Enable Blockiad
2. Enable metrics
3. Initiate a malicious tx from inside the wallet
4. Cancel/Reject the transaction
5. See that PPOM Metrics is sent in the logs

## **Screenshots/Recordings**

### **Before**

https://github.com/MetaMask/metamask-mobile/assets/54408225/12153f52-baeb-4344-b6f6-0c00c2a1d1fc

### **After**

https://github.com/MetaMask/metamask-mobile/assets/44811/efea6aa9-96f7-4b57-915c-5fe946daabdf

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
